frontend/views/containers/contributions/IncomeDetails.vue

```diff
@@ -216,7 +216,11 @@ export default ({
           data: {
             incomeDetailsType,
             [incomeDetailsType]: normalizeCurrency(this.form.amount),
-            paymentMethods
```

Comment:
This was the cause of the issue. when the user switches back to `pledger`, `paymentMethods` in its group profile was being reset to `[]`. updated the logic so that this field is not included in the payload unless necessary.